master/internal/scheduler/fitting.go

```diff
@@ -229,6 +198,8 @@ func findNotMultiSlotFit(
 		return nil
 	}
 
+	// For the single agent case, we want prioritize using the smallest
+	// available agents.
```

Comment:
Ah, yeah. Do you think that makes sense for the multi-slot case to still prefer small agents instead of going by the fitting function (I think that was the behavior previously)?

Reply:
My inclination is to just default to the fitting function on this, since I don't expect we will often have heterogenous instances.